db/sqldb/migrations/0003_domain_types.up.sql

```diff
@@ -0,0 +1,242 @@
+BEGIN;
+
+DROP TABLE user_account;
+
+DROP TYPE auth_provider;
+
+CREATE TYPE language AS ENUM (
+    'en',
+    'de',
+    'fr',
+    'es'
+);
+
+CREATE TYPE authn_mechanism AS ENUM (
+    'EMAIL_AND_PASS'
+);
+
+CREATE TABLE pacta_user (
+    id TEXT PRIMARY KEY,
+    authn_mechanism authn_mechanism NOT NULL,
+    authn_id TEXT NOT NULL,
+    -- What the user actually entered during sign up.
+    entered_email TEXT NOT NULL UNIQUE,
+    -- The validated, cleaned, consistently cased email address - excludes plus aliases, for example.
+    canonical_email TEXT NOT NULL UNIQUE,
+    admin BOOLEAN NOT NULL,
+    super_admin BOOLEAN NOT NULL,
+    name TEXT NOT NULL,
+    -- Null until the user explicitly choses something. Will be defaulted based on the domain accessed from.
+    preferred_language language,
+    created_at TIMESTAMPTZ NOT NULL,
+    UNIQUE(authn_mechanism, authn_id)
+);
+
+CREATE TABLE pacta_version (
+    id TEXT PRIMARY KEY,
+    name TEXT NOT NULL,
+    description TEXT NOT NULL,
+    digest TEXT NOT NULL UNIQUE,
+    created_at TIMESTAMPTZ NOT NULL,
+    -- NULL = is not default, TRUE = is default, enforced via the checks below.
+    is_default BOOLEAN
```

Comment:
does that mean `FALSE` isn't a valid value?? If so, a less confusing version would be `is_default BOOLEAN NOT NULL [DEFAULT FALSE]`

Reply:
Great question - I'll expand the language above, but essentially yes. It's a simple way of enforcing that at most one value is marked as default across all entities. If there was a unitary type (Null or Present) we could represent it that way, but there is not.

using `boolean not null` wouldn't work without imposing a global check that couldn't easily be enforced via an index.